src/proposer/block_builder.cpp

```diff
@@ -102,6 +102,7 @@ class BlockBuilderImpl : public BlockBuilder {
       const EligibleCoin &eligible_coin,
       const staking::CoinSet &coins,
       const CAmount fees,
+      const boost::optional<CScript> &coinbase_script,
```

Comment:
Should this really be optional?
I searched usages and found that `boost::none` value is only used in unit-tests. Which is strange, I would expect unit tests to test execution paths that are used in production.

Reply:
```
return proposer::GenerateBlock(*m_active_chain,
                                   *m_transaction_picker,
                                   *m_block_builder,
                                   *m_proposer_logic,
                                   wallet,
                                   coins,
                                   boost::none);
```
In `proposer.cpp` line 226.
The idea behind the script being optional is the fact that potentially we are gonna need different scripts to pay to different address types. At the moment i.e. this is used by the `proposetoaddress`.

Reply:
Ah, right. Sorry, missed this usage.